cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -397,13 +396,21 @@ func ValidateIPNetFromString(subnetStr string, minAddrs int64, isDualStack bool,
 			allErrs = append(allErrs, field.Invalid(fldPath, subnetStr, "couldn't parse subnet"))
 			return allErrs
 		}
-		numAddresses := ipallocator.RangeSize(svcSubnet)
+		numAddresses := rangeSizeBoundCheck(svcSubnet)
 		if numAddresses < minAddrs {
 			allErrs = append(allErrs, field.Invalid(fldPath, subnetStr, "subnet is too small"))
 		}
 	}
 	return allErrs
 }
+func rangeSizeBoundCheck(cidr *net.IPNet) int64 {
+	max := utilnet.RangeSize(cidr)
+	ones, bits := cidr.Mask.Size()
```

Comment:
same comment about commenting and possibly simplifying the logic limiting to 65536

Reply:
[1]